flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala

```diff
@@ -239,6 +241,196 @@ class CodeGenerator(
   }
 
   /**
+    * Generates a [[org.apache.flink.table.runtime.aggregate.AggregateHelper]] that can be passed
+    * to Java compiler.
+    *
+    * @param name       Class name of the Function. Must not be unique but has to be a valid Java
+    *                   class identifier.
+    * @param clazz      AggregateHelper Function to be generated.
+    * @param aggFields  position (in the input Row) of the input value for each aggregate
+    * @param aggregates List of all aggregate functions
+    * @param generator  code generator instance
+    * @param inputType  Input row type
+    *
+    * @tparam F Flink Function to be generated.
+    * @tparam T Return type of the Flink Function.
+    * @return instance of GeneratedFunction
+    */
+  def generateAggregateHelper[F <: Function, T <: Any](
```

Comment:
It seems that we are not using a lot of the tooling provided by `CodeGenerator`. It might make sense to implement the aggregation code gen independently of the `CodeGenerator`, but I'm not sure about that.